### DISCUSSION TITLE
单库场景下是否应该避免使用seata

### DISCUSSION THREAD

**Discussion by ali13378510bu** [apache/incubator-seata#4076] [Q&A] [answered]
请问一下 我们的项目是单库的架构，因为业务逻辑分割成多个微服务，在实现某些逻辑时涉及到链式微服务调用，从系统架构上来看我们是应该使用seata来实现分布式事物，还是应该将代码移植到这些微服务中，这样就可以避免使用分布式事物。您可以给出一些意见么，感谢。

在我看来：分布式事物回滚是有失败的可能，所以从系统可用性来看应该应该避免使用seata。 但是从微服务的设计上来看，重复的逻辑使用远程调用的方式显然更合理。

**Comment by funky-eyes** [comment, marked as answer]
如果只是单库，完全可以将业务分类后，在某个服务需要多表操作的，开启本地事务，通过这个服务直接发出修改其他表的sql，来保证一致性，避免分布式事务的出现。
